mu-plugins/central-hub/tests/phpunit/integration/config-store/getAllTestsStartingWith.php

```diff
@@ -0,0 +1,90 @@
+<?php
+/**
+ *  Tests for external API function getAllKeysStartingWith()
+ *
+ * @package    spiralWebDb\centralHub\Tests\Integration\ConfigStore
+ * @since      1.3.0
+ * @author     Robert A. Gadon
+ * @link       https://github.com/rgadon107/cornerstone
+ * @license    GNU General Public License 2.0+
+ */
+
+namespace spiralWebDb\centralHub\Tests\Integration\ConfigStore;
+
+use function KnowTheCode\ConfigStore\_the_store;
+use function KnowTheCode\ConfigStore\loadConfig;
+use function KnowTheCode\ConfigStore\getAllKeysStartingWith;
+use spiralWebDb\Cornerstone\Tests\Integration\Test_Case;
+
+/**
+ * Class Tests_GetAllKeysStartingWith
+ *
+ * @package spiralWebDb\centralHub\Tests\Integration\ConfigStore
+ * @group   config-store
+ */
+class Tests_GetAllKeysStartingWith extends Test_Case {
+
+	/**
+	 * Empty the store before starting these tests.
+	 */
+	public static function setUpBeforeClass() {
+		$store_keys = _the_store();
+		if ( empty( $store_keys ) ) {
+			return;
+		}
+
+		foreach ( array_keys( $store_keys ) as $store_key ) {
+			_the_store( $store_key, null, true );
+		}
+	}
+
+	/**
+	 * Test getAllKeysStartingWith() should filter and return all store keys that start with the given string.
+	 */
+	public function test_should_filter_and_return_all_store_keys_starting_with_given_string() {
+		$configs = [
+			'metabox.events'         => [
+				'Trinity Lutheran Church'  => 'Columbus, OH',
+				'Sanibel Episcopal Church' => 'Sanibel Island, FL',
+			],
+			'metabox.members'        => [
+				'Brandon Bird'     => 'First Trombone',
+				'Talia Marie Aull' => 'Soprano',
+			],
+			'shortcode.qa'           => [
+				'Question 1' => 'How many angels can dance on the head of a pin?',
+				'Question 2' => 'Is the moon made of green cheese?',
+			],
+			'custom_post_type.books' => [
+				'Title'  => 'The DaVinci Code',
+				'Author' => 'Dan Brown'
+			],
+		];
+		foreach ( $configs as $store_key => $config_to_store ) {
+			loadConfig( $store_key, $config_to_store );
+		}
+
+		$this->assertSame( [ 'metabox.events', 'metabox.members' ], getAllKeysStartingWith( 'metabox.' ) );
+		$this->assertSame( [ 'shortcode.qa' ], getAllKeysStartingWith( 'shortcode.' ) );
+		$this->assertSame( [ 'custom_post_type.books' ], getAllKeysStartingWith( 'custom_post_type.' ) );
+	}
```

Comment:
@rgadon107 You want to clean up at the end of this particular test to remove the configs that were added by this test.  Best Practice: clean as you go.

Reply:
@hellofromtonya Point well taken. Please see commit [0dc219f](https://github.com/rgadon107/cornerstone/pull/26/commits/0dc219f1b37f967450e291049a9081b0a363e0a0).

Reply:
@rgadon107 Actually, you know the keys that were loaded from `$configs`. Loop through those and remove.  It’s better not to invoke `setUpBeforeClass()`.  Why?  It’s purpose is to set up the class itself, ie before the tests run.

Reply:
@rgadon107 I moved this code, combined it, and refactored into a separate static function that can be reused.